src/vcpkg/binarycaching.cpp

```diff
@@ -2165,20 +2173,65 @@ namespace vcpkg
                     request.nuspec =
                         generate_nuspec(request.package_dir, action, m_config.nuget_prefix, m_config.nuget_repo);
                 }
+
+                m_remaining_packages_to_push++;
+                m_actions_to_push.push(ActionToPush{std::move(request), action.build_options.clean_packages});
+                return;
+            }
+        }
+        if (action.build_options.clean_packages == CleanPackages::YES)
+        {
+            m_fs.remove_all(action.package_dir.value_or_exit(VCPKG_LINE_INFO), VCPKG_LINE_INFO);
+        }
+    }
+
+    void BinaryCache::print_push_success_messages() { m_bg_msg_sink.print_published(); }
+
+    void BinaryCache::wait_for_async_complete()
```

Comment:
I think the function name is a bit misleading: If I understand this function correctly, it kills the background thread?